app/src/api/dataAccess/editHistory.ts

```diff
@@ -0,0 +1,88 @@
+import db from '../../db';
+import { EditHistoryEntry } from '../../frontend/models/edit-history-entry';
+import { DatabaseError } from '../errors/general';
+
+const baseQuery = `
+        SELECT
+            log_id as revision_id,
+            forward_patch,
+            reverse_patch,
+            date_trunc('minute', log_timestamp) as revision_timestamp,
+            username,
+            building_id
+        FROM logs
+        JOIN users ON logs.user_id = users.user_id`;
+
+export function getHistoryAfterId(id: string, count: number): Promise<EditHistoryEntry[]> {
+    /** 
+     * SQL with lower time bound specified (records after ID).
+     * The outer SELECT is so that final results are sorted by descending ID
+     * (like the other queries). The inner select is sorted in ascending order
+     * so that the right rows are returned when limiting the result set.
+     */
+    try {
+        return db.any(`
+            SELECT * FROM (
+                ${baseQuery}
+                WHERE log_id > $1
+                ORDER BY revision_id ASC
+                LIMIT $2
+            ) AS result_asc ORDER BY revision_id DESC`,
+            [id, count]
+        );
+    } catch(err) {
+        throw new DatabaseError(err);
+    }
```

Comment:
This and other cases of using `DatabaseError` is a test of handling database-related errors in a more consistent way. Wrapping the errors in our custom error class currently serves the only purpose of being able to distinguish, in the global error handler, between a database-related error and a different, generic server error (`pg-promise` does not have a single base class for errors as some are simply piped through from the driver - `node-postgres`)
In the case of other data access layer classes, the error handler might do some more complex error checking, for example to check if the error is actually due to the value of the arguments, or due to a connection problem.